frontend/pulse3d/src/serviceWorker.js

```diff
@@ -222,8 +225,15 @@ const interceptResponse = async (req, url) => {
       const resBodyToCheck = await response.json();
 
       // set the usage error to SW state to send in auth check, will return a 200 status
-      if (resBodyToCheck.error && resBodyToCheck.error === "UsageError")
+      if (resBodyToCheck.error && resBodyToCheck.error === "UsageError") {
         setUsageQuota(resBodyToCheck.message);
+      } else {
+        // most resent upload does not get read in time
```

Comment:
recent

Reply:
I think we want to avoid changing info from the server in the FE like this. Also if this is just to help keep the usage up to date in the FE, I think a better way of doing this would be to poll the usage every 10 seconds or so (in case another user logged in under that customer account does something that changes the usage), and also hit the usage route any time the new job route is successfully hit

Reply:
I don't think we'd need to poll, its initially set up to return a response with an error key if a limit is reached in the POST routes, that should be sufficient for keeping track. That's what we need to make sure still works with these changes.

Reply:
and we want to make sure that when job limits are reached, that the re-analysis tab and interactive tab are disabled
which is what the error response above triggers

Reply:
The front end does get restricted when usage limit is reached and I have also removed this from this code from the service worker. There will be "lag" between job creation and displaying credit usage on the front end, but users are always notified if a credit was used. A check in both front end and back end prevent the running of any jobs above the limits.